api/src/auth/utils/is-login-redirect-allowed.ts

```diff
@@ -1,12 +1,9 @@
 import { useEnv } from '@directus/env';
 import { toArray } from '@directus/utils';
-import { useLogger } from '../logger/index.js';
-import isUrlAllowed from './is-url-allowed.js';
+import { useLogger } from '../../logger/index.js';
+import isUrlAllowed from '../../utils/is-url-allowed.js';
 
-/**
- * Checks if the defined redirect after successful SSO login is in the allow list
- */
-export function isLoginRedirectAllowed(redirect: unknown, provider: string): boolean {
+export function isLoginRedirectAllowed(redirect: unknown, origin: string, provider: string): boolean {
```

Comment:
The `origin` param should be an optional option as it is expected to default to `PUBLIC_URL` origin. May be worth splitting the option into a host + protocol and stitching it together within the function.

```suggestion
export function isLoginRedirectAllowed(redirect: unknown, provider: string, options: IsLoginRedirectAllowedOptions): boolean {
```

Reply:
As the host is a required param my comment can be disregarded